### DISCUSSION TITLE
K selection in clustering

### DISCUSSION THREAD

**Discussion by khanetor** [alan-turing-institute/sktime#4687] [Q&A]
In clustering algorithms such as Kmeans or Kshape, is there a function to calculate the silhouette score to determine the best k value?

**Comment by fkiraly** [comment]
As far as I know, this does not exist in `sktime`, although it is very close to existing patterns for classification.

The `sktime` clusterers are compatible with `sklearn` if `numpy` based data containers are used - so if you can find the above implemented for `sklearn` clusterers, the naive combination with `sktime` clusters should, hopefully, work out of the box.

Would appreciate pointers to:

* implementations of the above for `sklearn` clusterers
* if you find these try, whether this works out of the box in combination for `sktime`
